### ISSUE TITLE
explicit bool conversion for c++11 streams breaks previously valid code

### ISSUE THREAD

**Issue by GoogleCodeExporter** [google/googletest#429]
```
In C++11, the standard streams do not have an `operator void*()`, but have an 
`explicit operator bool()` instead. This change breaks code like:

    std::stringstream ss;
    ASSERT_TRUE(ss);

In the expansion of `ASSERT_TRUE`, an `AssertionResult` is constructed from 
`ss`. Since it expects a boolean but `ss` requires an explicit conversion, 
compilation fails. While this is not caused by GoogleTest but rather by a 
change in the standard, it might be desirable to provide a workaround. I'm just 
highlighting the issue.
```

Original issue reported on code.google.com by `louis.di...@gmail.com` on 6 Apr 2013 at 3:33

**Comment by Klaim** [comment]
+1

Though I would have changed it to static_assert or 

    ::testing::AssertionResult(expression ? true : false)

**Comment by xmxwx** [comment]
The issue is already fixed since 8120f66c3249e253f03fdb48bee7e528bc038d31.
